### DISCUSSION TITLE
The Lifecycle of Niche Software

### DISCUSSION THREAD

**Discussion by Papa-26** [community/community#110207] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

GitHub is a miracle! The fact that commercial giants would sponsor an operation equally benefitting corporations and individual software developers seems to counter current economic theories. Github, as a universal development tool, facilitates the development of commercial megaprojects, down to highly specialized niche software.

But Open-Source, niche software faces particular problems beyond its conception, creation, and initial distribution due to its lack of commercial momentum. Successful scientific niche software tends to be developed in response to an acute need by scientists rather than software professionals. The initial distribution is facilitated by the specialist literature and peer exchanges, primarily motivated by the value of innovation. As time progresses and the digital ecosystem evolves, software maintenance lapses because nobody feels responsible anymore, and economic incentives are lacking. The particular product becomes more and more fragile, less and less functional, and eventually fades into the cloud of forgotten digital dust. When the initial need still exists, the corresponding wheel will be eventually reinvented, although at an additional cost that could have easily been prevented.

I was confronted with these issues in connection with a software tool I originally developed 15 years ago as a retirement project to help a colleague. At the time it was written in java 8 on a Windows platform, and was well received by the corresponding, narrow group of researchers. Over the years I extended the program's functionality, eased its progression through the various java language versions to number 21, extended its operation to the Linux and Mac platform, and incorporated Gradle as build tool. All could be achieved thanks to GitHub.

But as the java's LTS expiry date threatened to overtake my own, I started to look into ways of incorporating some 'immortality' into the project's repository. I invested a lot of work into software documentation, manuals, log file production and background information. I even learned, how to produce YouTube videos. However, I did miss guidance regarding what kind of information future users, and potential software professionals will need, to keep the basic ideas of the program functional. It would be nice to have a professional template for this purpose.

I hope to stimulate experts into contributing to this discussion, in the hope that future developers, who face similar issues, can profit from the collective wisdom.

**Comment by vistonviston** [comment]
If we talk about Java development, I can recommend [https://expert-soft.com/](https://expert-soft.com/) team. The guys worked with complex projects in the niches of commerce and healthcare, and are also ready to offer unique technological business solutions. They will help significantly increase the productivity of your company and implement automatization into your business processes.

**Comment by Papa-26** [comment]
OK, so we now have a link for developers who think that they need commercial help.

**Comment by silpa303** [comment]
A great viewpoint on open-source, niche software! But at the same time, keeping these tools relevant can be hard. Using containerization with Docker's help could probably be a nice idea. It ensures compatibility with the [latest technologies in the software industry](https://thinkpalm.com/blogs/top-5-technologies-that-will-rule-software-development-for-next-5-years/).

**Comment by Papa-26** [comment]
Thanks, something to keep in mind. However, at age 85 I am unlikely to ascend yet another, major technological step. I am busy enough with transforming my niche software from a Java basis to Python
